libs/storage/Tsavorite/cs/src/core/Index/Recovery/Recovery.cs

```diff
@@ -735,14 +787,20 @@ private void RecoverHybridLogFromSnapshotFile(long scanFromAddress, long recover
         {
             GetSnapshotPageRangesToRead(scanFromAddress, untilAddress, snapshotStartAddress, snapshotEndAddress, guid, out long startPage, out long endPage, out long snapshotEndPage, out int capacity, out var recoveryStatus, out int numPagesToReadFirst);
 
-            hlog.AsyncReadPagesFromDevice(startPage, numPagesToReadFirst, snapshotEndAddress,
-                                          hlog.AsyncReadPagesCallbackForRecovery,
-                                          recoveryStatus, recoveryStatus.recoveryDevicePageOffset,
-                                          recoveryStatus.recoveryDevice, recoveryStatus.objectLogRecoveryDevice);
-
-            for (long page = startPage; page < endPage; page += capacity)
+            for (long page = startPage; page < endPage; page += numPagesToReadFirst)
```

Comment:
Same comment regarding renaming numPagesToReadFirst to numPagesToReadPerIteration. When this was initially written, numPagesToReadFirst was correct; it was followed by one-at-a-time operations. Now we do batches on each iteration.